DDG4/edm4hep/EDM4hepFileReader.cpp

```diff
@@ -0,0 +1,320 @@
+//==========================================================================
+//  AIDA Detector description implementation
+//--------------------------------------------------------------------------
+// Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : A.Sailer
+//
+//==========================================================================
+
+/** \addtogroup Geant4EventReader
+ *
+ @{
+ \package EDM4hepFileReader
+ * \brief Plugin to read EDM4hep files
+ *
+ *
+ @}
+*/
+
+#include <CLHEP/Units/SystemOfUnits.h>
+
+#include <DDG4/EventParameters.h>
+#include <DDG4/Factories.h>
+#include <DDG4/Geant4InputAction.h>
+#include <DDG4/RunParameters.h>
+
+#include <edm4hep/EventHeaderCollection.h>
+#include <edm4hep/MCParticleCollection.h>
+
+#include <podio/Frame.h>
+#include <podio/ROOTReader.h>
+
+typedef dd4hep::detail::ReferenceBitMask<int> PropertyMask;
+
+
+namespace dd4hep::sim {
+
+  // we use the index of the objectID to identify particles
+  // we will not support MCParticles from different collections
+  using MCPARTICLE_MAP=std::map<int, int>;
+
+  /// get the parameters from the GenericParameters of the input EDM4hep frame and store them in the EventParameters
+  /// extension
+  template <class T=podio::GenericParameters> void EventParameters::ingestParameters(T const& source) {
+    auto const& intKeys = source.template getKeys<int>();
+    for(auto const& key: intKeys) {
```

Comment:
```suggestion
    for(auto const& key: source.template getKeys<int>()) {
```
Lifetime extension should work in this context, I think. (Similar for other types below)

Reply:
Done (to be pushed soon).